### PR TITLE
Appeals unsafe mode

### PR DESCRIPTION
**Pull request by caseywilliams**
## Summary

This will allow requests without a valid oauth token in appeals APIs if the following conditions are all met:

1. The app is running in development mode
2. The `APPEALS_API_UNSAFE_MODE` environment variable is set
3. The setting `modules_appeals_api.enable_unsafe_mode` is set

## Related issue(s)

None

## Testing done

Confirmed locally that token validity is ignored if and only if all three requirements are met

## What areas of the site does it impact?

Enables easier manual testing of non-auth-related features in local development

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/appeals_api/spec/factories/higher_level_reviews.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+require AppealsApi::Engine.root.join('spec', 'support', 'fixture_helpers.rb')
+
```
Comment (caseywilliams) [inline modules/appeals_api/spec/factories/higher_level_reviews.rb:4]: I am not sure why, but with this PR, I started seeing test failures [like these](https://github.com/department-of-veterans-affairs/vets-api/actions/runs/5293533052/jobs/9581966037). The error is from changes in #13001, which was merged yesterday, but I haven't seen this error locally or in other PRs until now. I'm hoping the second commit I pushed here will fix this for good.

Comment (michelpmcdonald) [reply modules/appeals_api/spec/factories/higher_level_reviews.rb:4]: I've been hitting unit test fails that I resolved with a require also

File: modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb [hunk 2]
```diff
@@ -55,6 +57,11 @@ def validate_auth_token!
 
     private
 
+    def unsafe_mode?
+      Rails.env.development? && \
```
Comment (cilestin) [inline modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb:61]: Any time I see rails environment checks like this I feel compelled to remind folk that all our server environments run rails production mode - even dev-api.va.gov. You'll need to use `Settings.vba_environment` if you want to run in unsafe mode on lower environments (e.g. `Settings.vsp_environment != 'production'`)

But if a rails environment check was what you intended, carry on! :)

File: modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb [hunk 3]
```diff
@@ -55,6 +57,11 @@ def validate_auth_token!
...
+        Settings.dig(:modules_appeals_api, :enable_unsafe_mode) && ENV['APPEALS_API_UNSAFE_MODE'].present?
```
Comment (cilestin) [inline modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb:62]: Where would this environment variable come from on deploy? I think we'd have to work with someone to make sure it's set on certain vsp environments.

Comment (caseywilliams) [reply modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb:62]: When we talked about this amongst the team ([here](https://jira.devops.va.gov/browse/API-27449)), our goal was to give ourselves some way to skip authentication while developing locally, but there was a lot of concern that the setting could bet turned on accidentally in the deployed environments. Requiring both a setting and an environment variable was an idea we settled on to make it extra difficult for this to happen

Comment (cilestin) [reply modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb:62]: Gotcha! Let's put some comments to that affect at the top of the method so folk without that context can know why it's checking things in triplicate :)

File: modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb (hunk 2 above)
Comment (kristen-brown) [reply modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb:61]: I believe it was intended for use in the local environment only (to make local testing easier).

File: modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb (hunk 3 above)
Comment (kristen-brown) [reply modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb:62]: I second @cilestin's comment about adding a comment to explain the purpose. 

Personally, I feel pretty confident with the dev environment + setting checks alone, but I also don't see the additional environment variable check as a bad thing for extra safety. (I also can't remember what my feedback was back when we discussed it as a team.) The only (small) downside that I see is that the developer hoping to skip the token validation step locally has to remember to set the config in two places.

**Review by kristen-brown — DISMISSED** [review]

**Comment by cilestin** [comment]
Could we get another look at this PR? It's been approved for weeks, and it hanging in limbo for so long makes me wonder just how necessary the feature is...

**Comment by kristen-brown** [comment]
> Could we get another look at this PR? It's been approved for weeks, and it hanging in limbo for so long makes me wonder just how necessary the feature is...

I had a use for it the other day, so I'd be good with us merging it if there are no other concerns!

File: modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb (hunk 3 above)
Comment (caseywilliams) [reply modules/appeals_api/app/controllers/concerns/appeals_api/openid_auth.rb]: I went ahead and removed the check for the environment variable here - it now only relies on running in dev + having the setting

**Review by kristen-brown — APPROVED** [review]

**Pull request merged**